api/app/notifications/notifications_sms_callback.py

```diff
@@ -5,11 +5,38 @@
 
 from app.errors import InvalidRequest, register_errors
 from app.notifications.process_client_response import validate_callback_data, process_sms_client_response
+from app.sap.oauth2 import require_oauth
 
 sms_callback_blueprint = Blueprint("sms_callback", __name__, url_prefix="/notifications/sms")
 register_errors(sms_callback_blueprint)
 
 
+@sms_callback_blueprint.route('/sap/<notification_id>', methods=['POST'])
+@require_oauth(scope=None)
+def process_sap_response(notification_id):
+    client_name = 'SAP'
+    data = json.loads(request.data)
```

Comment:
https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.get_json

Reply:
this change would expect that SAP is sending us correct Content-Type headers, otherwise it will return None. maybe they are, do we want to check?

Reply:
i just changed this. we can just test this in environment and see that it works. it most likely has the right header anyway. there's two other places in the codebase this is used, but i'm not changing them now (one for telstra and one for SNS)